p2p/src/authenticated/actors/tracker/record.rs

```diff
@@ -3,225 +3,501 @@ use commonware_cryptography::Verifier;
 use std::net::SocketAddr;
 
 /// Represents information known about a peer's address.
-#[derive(Clone)]
-pub enum Record<C: Verifier> {
+#[derive(Clone, Debug, PartialEq)]
+pub enum Address<C: Verifier> {
     /// Peer address is not yet known.
     /// Can be upgraded to `Discovered`.
-    /// Tracks the number of peer sets this peer is part of.
-    Unknown(usize),
+    Unknown,
+
+    /// Peer is the local node.
+    Myself(PeerInfo<C>),
 
     /// Provided during initialization.
     /// Can be upgraded to `Persistent`.
+    /// Will be tracked even if the peer is not part of any peer sets.
     Bootstrapper(SocketAddr),
 
     /// Discovered this peer's address from other peers.
-    /// Tracks the number of peer sets this peer is part of.
-    Discovered(usize, PeerInfo<C>),
+    Discovered(PeerInfo<C>),
 
-    /// Discovered this peer's address from other peers after it was bootstrapped.
-    /// Will continuously be tracked.
+    /// Discovered this peer's address from other peers after it was originally a bootstrapper.
+    /// Will be tracked even if the peer is not part of any peer sets.
     Persistent(PeerInfo<C>),
+
+    /// Peer is blocked.
+    /// We don't care to track its information.
+    Blocked,
+}
+
+/// Represents a record of a peer's address and associated information.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Record<C: Verifier> {
+    /// Address state of the peer.
+    address: Address<C>,
+
+    /// Number of peer sets this peer is part of.
+    sets: usize,
 }
 
 impl<C: Verifier> Record<C> {
-    /// Get the address of the peer.
-    ///
-    /// Returns None if the address is unknown.
-    pub fn get_address(&self) -> Option<SocketAddr> {
-        match &self {
-            Self::Unknown(_) => None,
-            Self::Bootstrapper(socket) => Some(*socket),
-            Self::Discovered(_, peer_info) => Some(peer_info.socket),
-            Self::Persistent(peer_info) => Some(peer_info.socket),
+    /// Create a new record with an unknown address.
+    pub fn unknown() -> Self {
+        Record {
+            address: Address::Unknown,
+            sets: 0,
         }
     }
 
-    /// Get the peer information if available.
-    pub fn get_peer_info(&self) -> Option<&PeerInfo<C>> {
-        match &self {
-            Self::Unknown(_) => None,
-            Self::Bootstrapper(_) => None,
-            Self::Discovered(_, peer_info) => Some(peer_info),
-            Self::Persistent(peer_info) => Some(peer_info),
+    /// Create a new record with the local node's information.
+    pub fn myself(peer_info: PeerInfo<C>) -> Self {
```

Comment:
nit: I'm not in love with `myself` for this. I wonder if `local` is more "in theme" with `bootstrapper`?

(I suppose we also consider IPs in the range 10.x as local, so this could cause some confusion...)